### ISSUE TITLE
Benchmarking program for unaligned accesses

### ISSUE THREAD

**Issue by nemequ** [nemequ/portable-snippets#8]
We need a program to benchmark the different methods for the [unaligned module](https://github.com/nemequ/portable-snippets/tree/master/unaligned).  Shouldn't be difficult now that the clock module is in reasonable shape…

@Cyan4973, did you do anything special for [that blog post](https://fastcompression.blogspot.fr/2015/08/accessing-unaligned-memory.html), or just benchmarking xxhash?  If the former, I don't suppose you still have the code sitting around somewhere (and would be willing to share it)?

**Comment by Cyan4973** [comment]
I guess I just used the internal benchmark module of `xxhsum` and `lz4` (command `-b`).

**Comment by travisdowns** [comment]
If you're on Linux x86, you can also consider [`uarch-bench`](https://github.com/travisdowns/uarch-bench) to test the "raw" performance of loads/stores of various size and misalignments, perhaps as a baseline to compare to the psnip versions. It measures all 64-byte alignments, and the results align with what we know from published performance and optimization manuals. In uses small snippets of asm for the actual test code, which is the only thing that would need to be ported to make it work on other archs.

**Comment by nemequ** [comment]
x86, and especially Linux, are pretty well tested.  So is ARM; this issue is really more for more exoctic architectures and compilers.

uarch-bench looks very cool, though; could be useful for [SIMDe](https://github.com/nemequ/simde/).

**Comment by travisdowns** [comment]
Right, it makes sense. I do want to support other mainstream archs on uarch bench, but that probably mostly just means ARM.
